// Copyright 2019 Google LLC
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

syntax = "proto2";

package icing.lib;

option java_package = "com.google.android.icing.proto";
option java_multiple_files = true;

// Encapsulates the configurations on how Icing should score and rank the search
// results.
// Next tag: 3
message ScoringSpecProto {
  // OPTIONAL: Indicates how the search results will be ranked.
  message RankingStrategy {
    enum Code {
      // No ranking strategy specified, documents will be returned in the
      // default order that the most recent document inserted into Icing comes
      // first.
      NONE = 0;

      // Ranked by user-provided document scores.
      DOCUMENT_SCORE = 1;

      // Ranked by document creation timestamps.
      CREATION_TIMESTAMP = 2;
    }
  }
  optional RankingStrategy.Code rank_by = 1;

  // OPTIONAL: Indicates the order of returned search results, the default is
  // DESC, meaning that results with higher scores come first. This order field
  // will be ignored if 'rank_by' is NONE.
  message Order {
    enum Code {
      // Search results will be returned in a descending order.
      DESC = 0;

      // Search results will be returned in a ascending order.
      ASC = 1;
    }
  }
  optional Order.Code order_by = 2;
}
